src/Prometheus.Client.AspNetCore/PrometheusExtensions.cs

```diff
@@ -48,10 +47,8 @@ void AddMetricsHandler(IApplicationBuilder coreapp)
 
                     using (var outputStream = response.Body)
                     {
-                        ScrapeHandler.Process(options.CollectorRegistryInstance, outputStream);
+                        await ScrapeHandler.ProcessAsync(options.CollectorRegistryInstance, outputStream);
```

Comment:
.ConfigureAwait(false) does not make any sense in asp .net core, so I removed it